proto/substrait/extended_expression.proto

```diff
@@ -19,33 +19,42 @@ message ExpressionReference {
     AggregateFunction measure = 2;
   }
   // Field names in depth-first order
+  //
+  // For primitive types, this will just be a single name. Nested types may
+  // have multiple names.
```

Comment:
```suggestion
  // For primitive types, this will just be a single name. Nested types will
  // have multiple names.
```
If a nested type then I think we'd need to name everything (i.e. don't allow partial name).  Though I may be wrong.